### PR TITLE
Add draft for service-vs-library pattern.

### PR DESCRIPTION
**Pull request by MaineC**
As requested in #89 this is a first draft of the service-vs-library pattern that I observed in DevOps teams.

Feedback, suggestions etc. welcome.

### CODE REVIEW

File: service-vs-library.md
Comment (NewMexicoKid) [inline service-vs-library.md]: typo: errros -> errors

**Comment by rrrutledge** [comment]
This writeup is good, Isabel.

File: service-vs-library.md
Comment (rrrutledge) [inline service-vs-library.md:54]: The **Patlet** talks about `deploying the same service in independent environments` yet the solution talks about factoring out shared service code into a library, which becomes the focus of collaboration (not the service).  I feel like these two sections of the document are incongruent with each other.  Are the teams collaborating on the source code for a library or a service?

**Comment by MaineC** [comment]
@NewMexicoKid Thanks for pointing out the typo.

@rrrutledge You're right - these two are incongruent. I've changed the Patlet to be better in line with the proposed solution. Deploying the same service twice with zero modification would presume that deployments are standardised across teams. While this may be true in some cases, more likely than not independent teams will have different deployment environments. Thank you for pointing out that flaw.

File: service-vs-library.md
Comment (rrrutledge) [inline service-vs-library.md:20]: Another way of getting around this problem is [30-day warranty](https://github.com/paypal/InnerSourcePatterns/blob/master/30-day-warranty.md).  Could be good to mention that somewhere in this document.

**Review by rrrutledge — COMMENTED** [review]
Seems pretty clear to me.  Just one comment.

Also, I accidentally made a commit (`cbb2bb2`) directly to this branch that I meant to make as a pull request suggestion to the branch.  If you don't like it then feel free to revert.

**Comment by MaineC** [comment]
@rrrutledge I didn't even know that committing to a branch suggested as PR would be possible for the maintainers of the original repository. Learnt something new there. Also, thanks for the corrections you included in your commits.

File: service-vs-library.md
Comment (rrrutledge) [reply service-vs-library.md:20]: Thanks.

**Review by rrrutledge — APPROVED** [review]
Looks good to me.

**Comment by MaineC** [comment]
@rrrutledge thanks for the approval. What does the remaining process for this pull request look like?

**Review by NewMexicoKid — APPROVED** [review]
I think this is solid enough to move into the accepted patterns in the main branch. We can pursue a second review at need if people reading the pattern open a PR with suggested changes.

**Pull request merged**
